components/scream/src/share/tests/field_tests.cpp

```diff
@@ -126,6 +126,47 @@ TEST_CASE("field", "") {
     // Trying to reshape into something that the allocation cannot accommodate should throw
     REQUIRE_THROWS (f1.get_reshaped_view<Pack<Real,32>***>());
   }
+
+  // Check whether we can correctly manipulate a field with two scalar types.
+  SECTION ("reshape and reduce with multiple scalar types") {
+    Field<Pack<Real,4>,Device> f1 (fid);
```

Comment:
This is fine, but notice that 1) in the FieldRepository all fields are `Field<Real,Device>` (no pack scalar type), and 2) the extent `fid.get_layout().dim(2)` will be interpreted as the number of packs, rather than the "physical" dimension of the field. Perhaps it's best to stick with `Field<Real,Device>`, to avoid create confusion in users.

Currently, we don't offer a way to change the scalar type of a Field (only the underlying view can be reshaped). I suspect this is not a limitation, since packs are needed at the computational level, where the whole Field structure is no longer needed (when calculations start, one should already be using the stored view (possibly reshaped), not the field).

Reply:
From this and your other comments, it sounds like I really don't understand everything that I need to in order to test reshaping of fields this way. Maybe we should go back to the drawing board and draft a test with specific inputs that produce specific expected outputs. That would actually allow me to document some of these things.

Reply:
Sure, and maybe I can write some better documentation.